dynawo/sources/Modeler/DataInterface/IIDM/DYNDataInterfaceIIDM.cpp

```diff
@@ -439,9 +439,11 @@ DataInterfaceIIDM::importVoltageLevel(IIDM::VoltageLevel& voltageLevelIIDM, cons
     //===========================
     IIDM::Contains<IIDM::Switch>::iterator itSwitch = voltageLevelIIDM.switches().begin();
     for (; itSwitch != voltageLevelIIDM.switches().end(); ++itSwitch) {
-      shared_ptr<SwitchInterface> sw = importSwitch(*itSwitch);  // in bus breaker topology, keep all switches
-      components_[sw->getID()] = sw;
-      voltageLevel->addSwitch(sw);
+      shared_ptr<SwitchInterface> sw = importSwitch(*itSwitch);
+      if (sw->getBusInterface1() != sw->getBusInterface2()) {  // if the switch is connecting one single bus, don't create a specific switch model
```

Comment:
Could we please add a dedicated unit test (if possible for both old and new libraries)?

Reply:
ok

Reply:
done!